src/core/ready.js

```diff
@@ -49,11 +47,28 @@ jQuery.extend( {
 			return;
 		}
 
-		// If there are functions bound, to execute
-		readyList.resolveWith( document, [ jQuery ] );
+		whenReady = function( fn ) {
+			readyCallbacks.push( fn );
+
+			if ( !++readyFiring ) {
+				while ( readyCallbacks.length ) {
+					fn = readyCallbacks.shift();
+					if ( jQuery.isFunction( fn ) ) {
+						fn.call( document, jQuery );
+					}
+				}
+			}
+
+			readyFiring--;
+		};
+
+		whenReady();
 	}
 } );
 
+// Make jQuery.ready promise compatible (gh-1778)
+jQuery.ready.then = jQuery.fn.ready;
```

Comment:
Yep, I agree with that.  Apparently I'm not as familiar with thenables as I thought.

FWIW part of my confusion may be attributed to something that seems weird in Chrome, but I have no idea if this is a legitimate thing.

If you run `Promise.resolve(jQuery.ready)`  then inspect the internal `[[PromiseValue]]` the returned promise, you can see that the value the promise was resolved with is not what was passed to the `.then` handler; the resolved value is `jQuery.ready` as I had initially expected.  Might be an implementation and/or devTools bug, but it doesn't look like it has any observable side effects other than confusion.